### DISCUSSION TITLE
xAI Support

### DISCUSSION THREAD

**Discussion by WuuBoLin** [karthink/gptel#462] [Ideas]
Is [xAI](https://x.ai/) supported? Or will it be supported?

**xAI**'s API [just become live](https://x.com/xai/status/1853505214181232828) not long ago with a **$25/mo in free credits till EOY**

**Comment by WuuBoLin** [comment]
The tweet mentioned it’s *"Compatible with OpenAI and Anthropic SDKs"*, so it doesn’t seem too complicated, right?

**Reply by karthink** [reply]
Yes, it should be straightforward then.  See the instructions in the README for Perplexity, Cerebras, Groq or any other OpenAI compatible API.  You just need to plug in the right values for xAI.

**Reply by karthink** [reply]
If you get it working, could you raise a PR to add the instructions to the README?  You can copy one of the other setup sections and tweak the code.

**Reply by WuuBoLin** [reply]
Sure! Thanks for the help.

**Reply by WuuBoLin** [reply]
It works as it expected! PR here: #466
